src/webui/api/appcontroller.cpp

```diff
@@ -1083,6 +1085,24 @@ void AppController::defaultSavePathAction()
     setResult(BitTorrent::Session::instance()->savePath().toString());
 }
 
+void AppController::getDirectoriesAction()
+{
+    requireParams({u"path"_s});
+
+    const QString path = params().value(u"path"_s);
+    const QString dir = path.left(path.lastIndexOf(u'/'));
+
+    QDirIterator it(dir, QDir::Dirs | QDir::NoDotAndDotDot | QDir::CaseSensitive);
```

Comment:
Wouldn't it be better to use `QDir::entryList()` instead?

Reply:
It doesn't matter that much, but using entryList makes code shorter.

I changed this code.